src/coreclr/tools/Common/TypeSystem/Common/Utilities/CustomAttributeTypeNameParser.cs

```diff
@@ -52,8 +57,28 @@ public static class CustomAttributeTypeNameParser
 
             TypeDesc type = resolver.Resolve(parsed);
 
-            typeWasNotFoundInAssemblyNorBaseLibrary = resolver._typeWasNotFoundInAssemblyNorBaseLibrary;
             return type;
+
+            static bool IsFullyQualified (TypeName typeName)
+            {
+                if (typeName.AssemblyName is null)
+                {
+                    return false;
+                }
+
+                if (typeName.IsConstructedGenericType)
```

Comment:
This does not seem to handle arrays/byrefs/pointers of generic types with unqualified argument.

Reply:
I meant type name like `G`1[[MyType]][], MyAssembly`. I still think that type names like that are not handled correctly here.

(The tests you have added are for type names like `G`1[[MyType[]]], MyAssembly`. I agree that those are handled correctly.)